### PR TITLE
Reset result before running a new test

### PR DESCRIPTION
**Pull request by jinjingforever**
This will basically clear the "pass" or "failed" label from the previous result when running a new test.

### CODE REVIEW

File: pkg/tests/lib/test-admin.js [hunk 1]
```diff
@@ -20,6 +20,10 @@ class TestAdmin extends Xen.Async {
     };
   }
   async onTestOptionClick(e) {
+    // Reset result.
+    this.state.result = undefined;
```
Comment (sjmiles) [inline pkg/tests/lib/test-admin.js:24]: The original intent was changing state will cause invalidation, so the typical pattern (for XenAsync) here is:
`this.mergeState({result: null});`

Also, I never assign 'undefined', only 'null'.

Comment (sjmiles) [reply pkg/tests/lib/test-admin.js:24]: Fwiw, I have a workstream to make XenPlus or something where the lifecycle is more like Particles.

**Review by sjmiles — APPROVED** [review]

**Comment by sjmiles** [comment]
Thank you, I was feeling stupid that I didn't just do it for you, then you did it. 👍

File: pkg/tests/lib/test-admin.js (hunk 1 above)
Comment (jinjingforever) [reply pkg/tests/lib/test-admin.js]: Ah thank you! Nice reminder on `mergeState`. Updated.

**Comment by jinjingforever** [comment]
Thank you!:)

**Pull request merged**